interface/src/app/node/node.js

```diff
@@ -92,5 +92,204 @@ angular.module('adage.node', [
   }]
 )
 
+.directive('highRangeExp', ['$http', '$log', function($http, $log) {
+  return {
+    templateUrl: 'node/high_range_exp.tpl.html',
+    restrict: 'E',
+    scope: {
+      nodeId: '@'
+    },
+    link: function($scope) {
+      $scope.queryStatus = 'Connecting to the server ...';
+      $scope.activities = {};
+      $scope.experiments = [];
+      $scope.top20Mode = true;
+      $scope.numExp = 20;
+
+      // Get activities that are related to the current node:
+      $http.get('/api/v0/activity/?limit=0&node=' + $scope.nodeId)
+        .then(function success(response) {
+          var sampleID;
+          for (var i = 0; i < response.data.objects.length; ++i) {
+            sampleID = response.data.objects[i].sample;
+            $scope.activities[sampleID] = response.data.objects[i].value;
+          }
+          return $http.get('/api/v0/experiment/?limit=0&node=' + $scope.nodeId);
```

Comment:
Can you pass parameters to `$http.get` instead of explicitly building a URL? Might make it a bit friendlier to read.